pkg/report/sarif_test.go

```diff
@@ -541,6 +541,176 @@ func TestReportWriter_Sarif(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "report with vulnerabilities and package locations",
+			input: types.Report{
+				ArtifactName: "debian:9",
+				ArtifactType: ftypes.ArtifactContainerImage,
+				Metadata: types.Metadata{
+					RepoTags: []string{
+						"debian:9",
+					},
+					RepoDigests: []string{
+						"debian@sha256:a8cc1744bbdd5266678e3e8b3e6387e45c053218438897e86876f2eb104e5534",
+					},
+				},
+				Results: types.Results{
+					{
+						Target: "library/test",
+						Class:  types.ClassOSPkg,
+						Packages: []ftypes.Package{
+							{
+								Name:    "foo",
+								Version: "1.2.3",
+								Locations: []ftypes.Location{
+									{
+										StartLine: 5,
+										EndLine:   10,
+									},
+									{
+										StartLine: 15,
+										EndLine:   20,
+									},
+								},
+							},
+						},
+						Vulnerabilities: []types.DetectedVulnerability{
+							{
+								VulnerabilityID:  "CVE-2020-0001",
+								PkgName:          "foo",
+								InstalledVersion: "1.2.3",
+								FixedVersion:     "3.4.5",
+								PrimaryURL:       "https://avd.aquasec.com/nvd/cve-2020-0001",
+								SeveritySource:   "redhat",
+								Vulnerability: dbTypes.Vulnerability{
+									Title:       "foobar",
+									Description: "baz",
+									Severity:    "HIGH",
+									VendorSeverity: map[dbTypes.SourceID]dbTypes.Severity{
+										vulnerability.NVD:    dbTypes.SeverityCritical,
+										vulnerability.RedHat: dbTypes.SeverityHigh,
+									},
+									CVSS: map[dbTypes.SourceID]dbTypes.CVSS{
+										vulnerability.NVD: {
+											V3Vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
+											V3Score:  9.8,
+										},
+										vulnerability.RedHat: {
+											V3Vector: "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
+											V3Score:  7.5,
+										},
+									},
+								},
+								Locations: []ftypes.Location{
+									{
+										StartLine: 15,
+										EndLine:   20,
+									},
+									{
+										StartLine: 30,
+										EndLine:   35,
+									},
```

Comment:
location from package is different- https://github.com/aquasecurity/trivy/blob/80bdd63cf850e7abc8e1101f12cfced798a4dd78/pkg/report/sarif_test.go#L46-L55

Reply:
You need to add locations in both places:
- Package
- DetectVulnerability
But these must be the same line numbers.

Reply:
It appears there might be referring to wrong test case in the comment. My changes actually begin from line 544. Nevertheless, I've refactored the test case for a package vulnerability as per your suggestion.